__tests__/app.test.js

```diff
@@ -306,10 +297,35 @@ describe('App', ()=>{
                 })
             })
         });
-        test('GET 404: should respond 404 when the endpoint does not exists', () => {
+    })
+    describe('GET /api/articles (topic query)', () => {
+        test('GET 200: should return an array of article objects filtered by topic when topic query is provided', () => {
             return request(app)
-            .get('/api/userz')
-            .expect(404) 
+            .get('/api/articles?topic=cats')
+            .expect(200)
+            .then((response) => {
+                const articles = response.body.articles
+                expect(Array.isArray(articles)).toBe(true)
+                articles.forEach((article) => {
+                    expect(article).toMatchObject({
+                        author: expect.any(String),
+                        title: expect.any(String),
+                        article_id: expect.any(Number),
+                        topic: 'cats',
+                        created_at: expect.any(String),
+                        votes: expect.any(Number),
+                        comment_count: expect.any(Number)
+                    })
+                })
+            })
+        });
+        test('GET 400: should respond with 400 for invalid topic query', () => {
+            return request(app)
+            .get('/api/articles?topic=123')
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe('topic not found')
+            })
```

Comment:
It would be good to add another test for Status 200, valid topic query, but has no articles responds with an empty array of articles, e.g. ?topic=paper.